src/VkDispatchTable.h

```diff
@@ -0,0 +1,4617 @@
+		fp_vkUpdateDescriptorSetWithTemplate(device, descriptorSet, descriptorUpdateTemplate, pData);
+	}
+	void getDescriptorSetLayoutSupport(const VkDescriptorSetLayoutCreateInfo* pCreateInfo, VkDescriptorSetLayoutSupport* pSupport) {
+		fp_vkGetDescriptorSetLayoutSupport(device, pCreateInfo, pSupport);
+	}
+	void cmdDrawIndirectCount(VkCommandBuffer commandBuffer, VkBuffer buffer, VkDeviceSize offset, VkBuffer countBuffer, VkDeviceSize countBufferOffset, uint32_t maxDrawCount, uint32_t stride) {
+		fp_vkCmdDrawIndirectCount(commandBuffer, buffer, offset, countBuffer, countBufferOffset, maxDrawCount, stride);
+	}
+	void cmdDrawIndexedIndirectCount(VkCommandBuffer commandBuffer, VkBuffer buffer, VkDeviceSize offset, VkBuffer countBuffer, VkDeviceSize countBufferOffset, uint32_t maxDrawCount, uint32_t stride) {
+		fp_vkCmdDrawIndexedIndirectCount(commandBuffer, buffer, offset, countBuffer, countBufferOffset, maxDrawCount, stride);
+	}
+	VkResult createRenderPass2(const VkRenderPassCreateInfo2* pCreateInfo, const VkAllocationCallbacks* pAllocator, VkRenderPass* pRenderPass) {
+		return fp_vkCreateRenderPass2(device, pCreateInfo, pAllocator, pRenderPass);
+	}
+	void cmdBeginRenderPass2(VkCommandBuffer commandBuffer, const VkRenderPassBeginInfo* pRenderPassBegin, const VkSubpassBeginInfo* pSubpassBeginInfo) {
+		fp_vkCmdBeginRenderPass2(commandBuffer, pRenderPassBegin, pSubpassBeginInfo);
+	}
+	void cmdNextSubpass2(VkCommandBuffer commandBuffer, const VkSubpassBeginInfo* pSubpassBeginInfo, const VkSubpassEndInfo* pSubpassEndInfo) {
+		fp_vkCmdNextSubpass2(commandBuffer, pSubpassBeginInfo, pSubpassEndInfo);
+	}
+	void cmdEndRenderPass2(VkCommandBuffer commandBuffer, const VkSubpassEndInfo* pSubpassEndInfo) {
+		fp_vkCmdEndRenderPass2(commandBuffer, pSubpassEndInfo);
+	}
+	void resetQueryPool(VkQueryPool queryPool, uint32_t firstQuery, uint32_t queryCount) {
+		fp_vkResetQueryPool(device, queryPool, firstQuery, queryCount);
+	}
+	VkResult getSemaphoreCounterValue(VkSemaphore semaphore, uint64_t* pValue) {
+		return fp_vkGetSemaphoreCounterValue(device, semaphore, pValue);
+	}
+	VkResult waitSemaphores(const VkSemaphoreWaitInfo* pWaitInfo, uint64_t timeout) {
+		return fp_vkWaitSemaphores(device, pWaitInfo, timeout);
+	}
+	VkResult signalSemaphore(const VkSemaphoreSignalInfo* pSignalInfo) {
+		return fp_vkSignalSemaphore(device, pSignalInfo);
+	}
+	VkDeviceAddress getBufferDeviceAddress(const VkBufferDeviceAddressInfo* pInfo) {
+		return fp_vkGetBufferDeviceAddress(device, pInfo);
+	}
+	uint64_t getBufferOpaqueCaptureAddress(const VkBufferDeviceAddressInfo* pInfo) {
+		return fp_vkGetBufferOpaqueCaptureAddress(device, pInfo);
+	}
+	uint64_t getDeviceMemoryOpaqueCaptureAddress(const VkDeviceMemoryOpaqueCaptureAddressInfo* pInfo) {
+		return fp_vkGetDeviceMemoryOpaqueCaptureAddress(device, pInfo);
+	}
+#if defined(VK_KHR_swapchain) && defined(VK_VERSION_1_0)
+#ifndef PFN_vkCreateSwapchainKHR_PROXY
```

Comment:
Why do these functions have guards for while the member functions above don't?
Similarly to when the functions are being loaded, these macro guards don't seem to be doing anything useful.

Reply:
The ones above are proxys build around 1.0 core, so they are guaranteed. Every other proxy is tied to an extension/require so it is guarded.